Mage/src/main/java/mage/abilities/mana/ManaOptions.java

```diff
@@ -562,42 +547,19 @@ public boolean removeEqualMana(Mana manaToRemove) {
         return result;
     }
 
-    public static void addManaCombination(Mana mana, Mana existingMana, List<Mana> payCombinations, List<String> payCombinationsStrings) {
-        Mana newMana = existingMana.copy();
-        newMana.add(mana);
-        payCombinations.add(newMana);
-        payCombinationsStrings.add(newMana.toString());
-    }
-
-    public void removeDuplicated() {
-        Set<String> list = new HashSet<>();
 
-        for (int i = this.size() - 1; i >= 0; i--) {
-            String s;
-            if (this.get(i) instanceof ConditionalMana) {
-                s = this.get(i).toString() + ((ConditionalMana) this.get(i)).getConditionString();
-            } else {
-                s = this.get(i).toString();
-            }
-            if (s.isEmpty()) {
-                this.remove(i);
-            } else if (list.contains(s)) {
-                // remove duplicated
-                this.remove(i);
-            } else {
-                list.add(s);
-            }
-        }
-
-        // Remove fully included variations
-        // TODO: research too many manas and freeze (put 1 card to slow down, put 3 cards to freeze here)
-        //  battlefield:Human:Cascading Cataracts:1
+    /**
+     * Remove fully included variations. If both {R} and {R}{W} are in this, then {R} will be removed.
+     * TODO: Why is this needed
```

Comment:
If it's a performance optimization, it may not be worth the computational cost of doing this operation on a Set